mod_config/controllers.py

```diff
@@ -288,29 +288,38 @@ def services():
         # Process uploaded file
         file = request.files[form.file.name]
         if file:
-            filename = secure_filename(file.filename)
             temp_path = os.path.join('./pipot/services/temp', filename)
             final_path = os.path.join('./pipot/services', filename)
             if not os.path.isfile(final_path):
                 file.save(temp_path)
-                # Import and verify module
-                try:
-                    instance = ServiceLoader.load_from_file(temp_path)
-                    # Auto-generate tables
-                    instance.get_used_table_names()
-                    # Move
-                    os.rename(temp_path, final_path)
-                    service = Service(instance.__class__.__name__,
-                                      form.description.data)
-                    g.db.add(service)
-                    g.db.commit()
-                    # Reset form, all ok
-                    form = NewServiceForm(None)
-                except ServiceLoader.ServiceLoaderException as e:
-                    # Remove file
-                    os.remove(temp_path)
-                    # Pass error to user
-                    form.errors['file'] = [e.value]
+                filenames = []
+                filename = secure_filename(file.filename)
+                if zipfille.is_zipfile(filename):
+                    with zipfile.ZipFile(containe, 'r') as zip_ref:
+                        zip_ref.extractall(temp_path)
+                        file_names.extend(zip_ref.namelist())
+                        zip_ref.close()
+                else:
+                    filenames.append(filename)
+                for filename in filenames:
+                    # Import and verify module
```

Comment:
Did you try this out? If I'm not mistaken, this will throw a ton of errors if you upload a zip with multiple files in it, of which only a single one is the actual service.